satpy/readers/generic_image.py

```diff
@@ -55,14 +94,42 @@ def start_time(self):
     def end_time(self):
         return self.finfo['end_time']
 
-    def get_dataset(self, key, info, out=None):
+    def get_dataset(self, key, info):
         """Get a dataset from the file."""
-
-        logger.debug("Reading %s.", key.name)
-        values = self.file_content[key.name]
-        selected = np.array(values)
-        out = np.rot90(np.fliplr(np.transpose(selected)))
-        info['filename'] = self.finfo['filename']
-        ds = Dataset(out, copy=False, **info)
-        return ds
-
+        logger.debug("Reading %s.", key)
+        return self.file_content[key.name]
+
+
+def get_geotiff_area_def(filename, crs):
+    """Read area definition from a geotiff."""
+    from osgeo import gdal
+    from pyresample.geometry import AreaDefinition
+    from pyresample.utils import proj4_str_to_dict
+    fid = gdal.Open(filename)
```

Comment:
Don't `rasterio`'s CRS objects have this functionality (pulling the geotransform/etc out of the file)? Or the rasterio objects themselves? I thought there was a `to_rasterio` method but I don't see it. There is an example of getting the Affine object which is the same as the geotransform is purpose: http://xarray.pydata.org/en/stable/generated/xarray.open_rasterio.html

Reply:
The extents could be derived from the rasterio attributes it seems, but there is the PCS ID that needs to be read from the geotiff metadata not available via rasterio. One option would be to use `my_area` instead, but then the definition in the files would not be identical.

Reply:
The `rasterio`'s `.crs`attribute is just a string, e.g. `'+init=epsg:3035'`.